### PR TITLE
Ocp 269 Allow to install wix and wine on ubuntu 16.04 and Debian Stretch

### PR DESCRIPTION
**Pull request by dabrain34**

### CODE REVIEW

**Comment by turran** [comment]
The commit 65495ee and b4d020d shouldnt be included right? Those are already part of master

File: recipes/mono/0008-mono-os-mutex-mingw32-cross.patch [hunk 1]
```diff
@@ -2,10 +2,10 @@
 +++ mono-4.6.0/mono/utils/mono-os-mutex.h	2017-06-14 13:24:43.654620494 +0200
 @@ -154,7 +154,7 @@
  #else
- 
+
  /* Vanilla MinGW is missing some defs, load them from MinGW-w64. */
 -#if defined __MINGW32__ && !defined __MINGW64_VERSION_MAJOR && (_WIN32_WINNT >= 0x0600)
 +#if defined __MINGW32__
```
Comment (turran) [inline recipes/mono/0008-mono-os-mutex-mingw32-cross.patch]: Why this?

File: recipes/mono/0008-mono-os-mutex-mingw32-cross.patch [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
- 
- /* Fixme: Opaque structs */
+
+/bin/bash: q: command not found
```
Comment (turran) [inline recipes/mono/0008-mono-os-mutex-mingw32-cross.patch]: What :)?

File: recipes/mono.recipe [hunk 3]
```diff
@@ -13,10 +13,10 @@ class Recipe(recipe.Recipe):
                         ' --disable-dtrace ' \
                         ' --with-mcs-docs=no '
     config_sh = 'sh autogen.sh'
-    files_bins = ['mono-sgen', 'mono']
     platform_files_bins = {
-        Platform.DARWIN: ['xbuild'],
-        Platform.LINUX: ['xbuild'],
+        Platform.DARWIN: ['mono-sgen', 'mono', 'xbuild'],
+        Platform.LINUX: ['mono-sgen', 'mono', 'xbuild'],
+        Platform.WINDOWS: ['mono-sgen', 'mono']
```
Comment (turran) [inline recipes/mono.recipe]: This has been already fixed on master, why going back to the previous change?

File: packages/gstreamer-1.0-sdk/gstreamer-1.0-sdk.package [hunk 4]
```diff
@@ -8,7 +8,7 @@ class SDKPackage(package.SDKPackage):
     longdesc = "GStreamer 1.0 SDK"
     title = "GStreamer 1.0 SDK"
     url = "http://www.gstreamer.com"
-    version = '2013.6'
+    version = '1.0.0'
```
Comment (turran) [inline packages/gstreamer-1.0-sdk/gstreamer-1.0-sdk.package]: Why changing the version?

File: cerbero/utils/__init__.py [hunk 5]
```diff
@@ -317,7 +324,7 @@ def escape_path(path):
 
 def get_wix_prefix():
     if 'WIX' in os.environ:
-        wix_prefix = os.path.join(os.environ['WIX'], 'bin')
+        wix_prefix = os.path.join(os.environ['WIX'])
```
Comment (turran) [inline cerbero/utils/__init__.py]: What is this joining?

File: cerbero/utils/__init__.py [hunk 6]
```diff
@@ -105,6 +105,13 @@ def to_winepath(path):
         return path
 
 
+def from_winepath(path):
```
Comment (turran) [inline cerbero/utils/__init__.py]: This method has been discarded on previous merge, just keep a tmp of the original file and remove it, the current code has the example of the usage

File: cerbero/packages/wix_packager.py [hunk 7]
```diff
@@ -306,7 +335,10 @@ def _create_bundle(self, config_path, paths):
 
         # Clean up
         if not self.keep_temp:
-            os.remove(sources[0])
+            source_0 = sources[0]
+            if self._with_wine:
+                source_0 = from_winepath(source_0)
```
Comment (turran) [inline cerbero/packages/wix_packager.py]: Follow the same approach as it is right now

File: cerbero/packages/wix_packager.py [hunk 8]
```diff
@@ -243,11 +266,17 @@ def _create_msi(self, config_path):
...
```
Comment (turran) [inline cerbero/packages/wix_packager.py]: Use the same approach as it is right now

File: cerbero/packages/wix_packager.py [hunk 9]
```diff
@@ -243,11 +266,17 @@ def _create_msi(self, config_path):
...
+            os.remove(source_0)
             for f in wixobjs:
-                os.remove(f)
+                real_f = f
+                if self._with_wine:
+                    real_f = from_winepath(real_f)
```
Comment (turran) [inline cerbero/packages/wix_packager.py]: Use the same approach as it is right now

File: cerbero/packages/wix_packager.py [hunk 10]
```diff
@@ -103,18 +111,28 @@ def create_merge_module(self, output_dir, package_type, force, version,
...
-            os.remove(sources[0])
-            for f in wixobjs:
-                os.remove(f)
-                try:
-                    os.remove(f.replace('.wixobj', '.wixpdb'))
-                except:
-                    pass
+            source_0 = sources[0]
+            if self._with_wine:
+                source_0 = from_winepath(source_0)
```
Comment (turran) [inline cerbero/packages/wix_packager.py]: Use the same approach as it is right now

File: cerbero/packages/wix_packager.py [hunk 11]
```diff
@@ -103,18 +111,28 @@ def create_merge_module(self, output_dir, package_type, force, version,
...
+            os.remove(source_0)
+            if not self.package.wix_use_fragment:
+              for f in wixobjs:
+                  real_f = f
+                  if self._with_wine:
+                      real_f = from_winepath(real_f)
```
Comment (turran) [inline cerbero/packages/wix_packager.py]: Use the same approach as it is right now

File: cerbero/bootstrap/linux.py [hunk 12]
```diff
@@ -19,21 +19,52 @@
...
+            self.winetricks_tool = os.path.join(self.config.build_tools_prefix, 'winetricks')
+            WINETRICKS_URL = 'https://raw.githubusercontent.com/Winetricks/winetricks/master/src/winetricks'
+            if not os.path.isfile(self.winetricks_tool):
+                shell.download(WINETRICKS_URL, self.winetricks_tool)
+            shell.call ('chmod +x %s' % self.winetricks_tool)
+        else:
+            self.winetricks_tool = 'winetricks'
+            shell.call(self.tool % ' winetricks')
+
+    def _download_wine_package(self, path_name, package_name):
```
Comment (marecki315) [inline cerbero/bootstrap/linux.py]: Is this method used anywhere?

File: cerbero/bootstrap/linux.py [hunk 13]
```diff
@@ -19,21 +19,52 @@
...
+        package_path = os.path.expanduser('~/.cache/winetricks/%s/' % path_name)
+        url = 'ftp://fluendosys:fluendo\ sys@officestorage1.fluendo.lan/data/fluendo/tech/private/cerbero/custom_packages/wine/%s' % package_name
```
Comment (marecki315) [inline cerbero/bootstrap/linux.py]: I wouldn't trust officestorage1

File: packages/gstreamer-1.0-sdk/gstreamer-1.0-sdk.package (hunk 4 above)
Comment (turran) [reply packages/gstreamer-1.0-sdk/gstreamer-1.0-sdk.package]: This has not been solved yet ...

File: cerbero/bootstrap/linux.py (hunk 13 above)
Comment (turran) [reply cerbero/bootstrap/linux.py]: We should :)

File: cerbero/packages/wix.py [hunk 14]
```diff
@@ -67,11 +67,23 @@ def __init__(self, config, package):
         self.platform = config.platform
         self.target_platform = config.target_platform
         self._with_wine = self.platform != Platform.WINDOWS
-        self.prefix = config.prefix
+        if self._with_wine:
+          self.wix_wine_prefix = '/tmp/flu_cwwp'
+          if os.path.exists(self.wix_wine_prefix):
+            os.remove(self.wix_wine_prefix)
```
Comment (marecki315) [inline cerbero/packages/wix.py]: This will raise an exception if self.wix_wine_prefix is a directory

File: cerbero/packages/wix.py [hunk 15]
```diff
@@ -67,11 +67,23 @@ def __init__(self, config, package):
...
+        else:
+          self.prefix = config.prefix
         self.filled = False
         self.id_count = 0
         self.ids = {}
 
+    def __del__(self):
+        if self._with_wine:
+          if os.path.exists(self.wix_wine_prefix):
+            os.remove(self.wix_wine_prefix)
```
Comment (marecki315) [inline cerbero/packages/wix.py]: Same as above

File: cerbero/packages/wix.py [hunk 16]
```diff
@@ -139,6 +151,79 @@ def _format_version(self, version):
...
+    def _add_file(self, filepath):
+        dirpath, filename = os.path.split(filepath)
+        self._add_directory(dirpath)
+        dirid = self._dirids[dirpath]
+
+        component = etree.SubElement(self.component_group, 'Component',
+            Id=self._format_path_id(filepath), Guid=self._get_uuid(), Directory=dirid)
+        print(filepath)
+        print(dirid)
+        #import pdb;pdb.set_trace()
```
Comment (marecki315) [inline cerbero/packages/wix.py]: Not needed I suppose

**Pull request merged**